kura/examples/bnd_examples/org.eclipse.kura.demo.bnd.heater/src/main/java/org/eclipse/kura/demo/bnd/heater/Heater.java

```diff
@@ -0,0 +1,305 @@
+/*******************************************************************************
+ * Copyright (c) 2011, 2017 Eurotech and/or its affiliates
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech
+ *******************************************************************************/
+
+package org.eclipse.kura.demo.bnd.heater;
+
+import java.util.Date;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.kura.cloud.CloudClient;
+import org.eclipse.kura.cloud.CloudClientListener;
+import org.eclipse.kura.cloud.CloudService;
+import org.eclipse.kura.configuration.ConfigurableComponent;
+import org.eclipse.kura.message.KuraPayload;
+import org.osgi.service.component.annotations.*;
+import org.osgi.service.metatype.annotations.Designate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.osgi.service.component.ComponentException;
+
+/**
+ * The Heater demo using annotations
+ * <p>
+ * <strong>Note:</strong> The component must be marked as
+ * {@code immediate = true} and {@code configurationPolicy = REQUIRE}.
+ * </p>
+ *
+ * @author David Woodard
+ */
+@Designate(ocd = HeaterConfig.class)
+@Component(
+        immediate = true,
+        configurationPolicy = ConfigurationPolicy.REQUIRE,
+        property = "service.pid=org.eclipse.kura.demo.bnd.heater.Heater"
+)
+public class Heater implements ConfigurableComponent, CloudClientListener {
+
+    private static final Logger s_logger = LoggerFactory.getLogger(Heater.class);
```

Comment:
Yep, will make a commit shortly. I was using IntelliJ to make sure I was avoiding any Eclipse specific tooling. May be nice to include an IntelliJ code formatter that mimics what we have for Eclipse.